src/qt/bitcoingui.cpp

```diff
@@ -529,7 +529,7 @@ void BitcoinGUI::createActions()
 
                 connect(action, &QAction::triggered, [this, path] {
                     auto activity = new OpenWalletActivity(m_wallet_controller, this);
-                    connect(activity, &OpenWalletActivity::opened, this, &BitcoinGUI::setCurrentWallet);
+                    connect(activity, &OpenWalletActivity::opened, this, &BitcoinGUI::setCurrentWallet, Qt::QueuedConnection);
```

Comment:
665: pls consider rebasing after #6648 and adding https://github.com/bitcoin-core/gui/pull/696 as a follow-up